Tools/SwiftBrowser/Source/BrowserViewModel.swift

```diff
@@ -67,6 +137,38 @@ final class BrowserViewModel {
         }
     }
 
+    // MARK: Dialog properties
+
+    var isPresentingDialog = false {
+        didSet {
+            if !isPresentingDialog {
+                currentDialog = nil
+            }
+        }
+    }
+
+    @ObservationIgnored
```

Comment:
Why ignored?

Reply:
I figured that it wasn't necessary since the "isPresentingDialog" property takes care of the observable-ness needed

Reply:
I'd personally avoid overeager optimizations. If you capture two dependencies and they both change in one render cycle that's still one re-render. Conversely if you only capture one and the other changes you'll have a correctness bug.